velox/expression/ExprRewriteRegistry.cpp

```diff
@@ -28,6 +28,15 @@ void ExprRewriteRegistry::clear() {
 
 core::TypedExprPtr ExprRewriteRegistry::rewrite(
     const core::TypedExprPtr& expr) {
+  // Do not rewrite expression if all inputs are constant.
+  const auto& inputs = expr->inputs();
+  if (std::all_of(
+          inputs.begin(), inputs.end(), [](const core::TypedExprPtr& input) {
+            return input->isConstantKind();
+          })) {
+    return expr;
```

Comment:
Should this be a CHECK ? The caller should not call this API with all constant inputs.

Reply:
Thanks for the suggestion, this is called by `ExprOptimizer` and by `ExprCompiler`. Adding this as a `VELOX_CHECK` would cause expression compilation to fail for expressions with all constant inputs when `ExprSet` is constructed with `enableConstantFolding=false`. Could I modify this assignment in `ExprCompiler.cpp` from
```
  auto rewritten = enableConstantFolding
      ? expression::optimize(expr, queryCtx, pool)
      : expression::ExprRewriteRegistry::instance().rewrite(expr);
```
to 
```
  auto rewritten = enableConstantFolding ? expression::optimize(expr, queryCtx, pool) : expr;
```

Reply:
Do you know why we need 'enableConstantFolding' flag? Where do we set it to 'false'?

If we change the meaning of the flag, then we may need to rename it as well. 

CC: @pedroerp @bikramSingh91

Reply:
@pramodsatya Let's create a PR to remove the flag and see what breaks.